salt/states/firewalld.py

```diff
@@ -153,37 +154,85 @@ def __virtual__():
 
 def present(name,
             block_icmp=None,
+            prune_block_icmp=False,
             default=None,
             masquerade=False,
             ports=None,
+            prune_ports=False,
             port_fwd=None,
+            prune_port_fwd=False,
             services=None,
-            prune_services=True,
+            prune_services=False,
```

Comment:
What effect is this going to have on people who upgrade Salt with existing `firewalld` states?

Reply:
Sorry about the delayed response, I realized I had email notifications turned off. 

Changing the default of `prune_services` would likely affect some people who upgrade. If they did not specify `prune_services: True`, then any configurations added in `firewalld.present` would be added without removing existing services using this version of `firewalld.present`. It also wouldn't prune the other existing firewalld configurations (e.g. rich rules, ports, etc), which it previously would do regardless of the value of `prune_services`. 

However, the old behavior of the function required a complete definition of all preexisting firewall configurations along with the few things that you'd want to change, making it unusable unless you declared all the firewall configurations in a single location. The new behavior when setting these defaults to `True` only requires that you list the configurations you care to add, optionally pruning preexisting configurations only if specified.

We could revert it to its previous default of `False`. To keep the new `prune_` parameters consistent, those should probably be `False` by default as well. However, this would lead to more verbose state functions when you simply want to add services, rich rules, etc without affecting currently present items. 

Do you have any suggestions on how to add the new functionality that supports modular firewall configurations while leaving existing implementations of the function unaffected?

Reply:
Hey @cachedout, just wanted to follow up. Any suggestions on the above comments?

Reply:
We should not just outright change this, because people are using this as is.

We could change it to None, and if it is None, log a deprecation message that says it defaults to True, but that will be changed to False in 2 releases in the future.